codecov_auth/commands/owner/interactors/update_self_hosted_settings.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass
+
+from django.conf import settings
+
+import services.self_hosted as self_hosted
+from codecov.commands.base import BaseInteractor
+from codecov.commands.exceptions import Unauthenticated, ValidationError
+from codecov.db import sync_to_async
+from services.refresh import RefreshService
+
+
+@dataclass
+class UpdateSelfHostedSettingsInput:
+    auto_activate_members: bool = False
+
+
+class UpdateSelfHostedSettingsInteractor(BaseInteractor):
+    def validate(self) -> None:
+        if not self.current_user.is_authenticated:
+            raise Unauthenticated()
+
+        if not settings.IS_ENTERPRISE:
+            raise ValidationError(
+                "enable_autoactivation and disable_autoactivation are only available in self-hosted environments"
```

Comment:
nit: enable_auto_activation vs. enable_autoactivation

Reply:
I could see it either way, but because we're camelcasing the input on AutoActivate, it seems more consistent

Reply:
these are what the methods are called in `self_hosted.py`

Reply:
ooo okay makes sense